libbindgen/src/ir/context.rs

```diff
@@ -938,11 +938,9 @@ impl<'ctx> BindgenContext<'ctx> {
                     return true;
                 }
 
-                let name = item.canonical_name(self);
+                let name = item.real_canonical_name(self, false, true);
                 match *item.kind() {
-                    ItemKind::Module(..) => {
-                        self.options().enable_cxx_namespaces
-                    }
+                    ItemKind::Module(..) => true,
```

Comment:
So will this mean that if we have `--whitelist-type Foo` we will still have `namespace bar` as a module in the generated bindings?

```c++
struct Foo { };

namespace bar {
    struct Baz { };
}
```

Any way we can avoid this? It seems like we would need a bottom up traversal phase where we mark modules which contain any whitelisted items.